src/vconsole/vconsole-setup.c

```diff
@@ -223,51 +223,85 @@ static int font_load_and_wait(const char *vc, const char *font, const char *map,
  * processing, stty iutf8
  */
 static void setup_remaining_vcs(int fd, int vc_max, bool utf8) {
+        struct console_font_op cfo;
         struct vt_stat vcs = {};
+        struct unimapinit adv = {};
         struct unimapdesc unimapd;
         _cleanup_free_ struct unipair* unipairs = NULL;
-        int i, r;
+        _cleanup_free_ unsigned char *fontbuf = NULL;
+        const char *err_ioctl;
+        int i, r, fd_d;
 
         unipairs = new(struct unipair, USHRT_MAX);
         if (!unipairs) {
                 log_oom();
                 return;
         }
 
+        fontbuf = new(unsigned char, 32*32*512/8);
+        if (!fontbuf) {
+                log_oom();
+                return;
+        }
+
         /* get active, and 16 bit mask of used VT numbers */
         r = ioctl(fd, VT_GETSTATE, &vcs);
         if (r < 0) {
                 log_warning_errno(errno, "VT_GETSTATE failed, ignoring remaining consoles: %m");
                 return;
         }
 
-        for (i = 1; i <= vc_max; i++) {
-                int vcfd;
-                struct console_font_op cfo = {};
+        /* get fonts from source console */
+        cfo.op = KD_FONT_OP_GET;
+        cfo.flags = 0;
+        cfo.width = cfo.height = 32;
+        cfo.charcount = 512;
+        cfo.data = fontbuf;
 
-                vcfd = -1;
-                if (i == vcs.v_active || !is_allocated(vcfd, i) || !is_settable(&vcfd, i))
+        r = ioctl(fd, KDFONTOP, &cfo);
+        if (r < 0) {
+                err_ioctl = "KD_FONT_OP_GET";
+        } else {
+                unimapd.entries  = unipairs;
+                unimapd.entry_ct = USHRT_MAX;
+                r = ioctl(fd, GIO_UNIMAP, &unimapd);
+                if (r < 0)
+                        err_ioctl = "GIO_UNIMAP";
+                else
+                        cfo.op = KD_FONT_OP_SET;
+        }
+        if (r < 0)
+                (void) log_warning_errno(errno, "%s failed, fonts will not be copied: %m", err_ioctl);
+
+        for (i = 1; i <= vc_max; i++) {
+                fd_d = -1;
+                if (i == vcs.v_active || !is_allocated(fd_d, i) || !is_settable(&fd_d, i))
                         continue;
 
-                (void) toggle_utf8(vcfd, utf8);
+                (void) toggle_utf8(fd_d, utf8);
 
-                /* copy font from active VT, where the font was uploaded to */
-                cfo.op = KD_FONT_OP_COPY;
-                cfo.height = vcs.v_active-1; /* tty1 == index 0 */
-                (void) ioctl(vcfd, KDFONTOP, &cfo);
+                r = 0;
+                if (cfo.op != KD_FONT_OP_SET || (r = ioctl(fd_d, KDFONTOP, &cfo)) < 0) {
+                        if (r < 0)
+                                (void) log_warning_errno(errno, "KD_FONT_OP_SET failed, fonts will not be copied to tty%i: %m", i);
+                        (void) close(fd_d);
+                        continue;
```

Comment:
The logic here is too convoluted... In general your changes simplify the code, but here you're making stuff really hard to follow. Maybe separate out the test for `cfo.op != KD_FONT_OP_SET`:
```
 if (cfo.op != KD_FONT_OP_SET)
   continue;
```